server/routes/signup.js

```diff
@@ -0,0 +1,27 @@
+let express = require('express');
+let router = express.Router();
+const passport = require('passport')
+
+const initPassport = require('../passport-config')
+initPassport(passport)
+
+const User = require('../models/user')
+
+router.route('/')
+  .post((req, res, next) => {
+    if (req.body.password.length >= 6 && req.body.firstName !== null && req.body.lastName !== null) {
+    User.register(new User({firstName: req.body.firstName, lastName:req.body.lastName, email: req.body.email}), req.body.password)
```

Comment:
What is this function doing? Does the password get hashed and saved to the database?

Reply:
Yes, that's exactly right. The other fields are saved normally, the password is hashed and hash and salt values are saved to the DB.